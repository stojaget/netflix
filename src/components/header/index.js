import React, {useState} from 'react';
import {Link as ReactRouterLink} from 'react-router-dom';
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Profile,
  Dropdown,
  SearchIcon,
  Search,
  SearchInput,
  PlayButton,
} from "./styles/header";

export default function Header({ bg=true, children, ...restProps }){
    return bg ? <Background {...restProps} data-testid="header-bg">{children}</Background> : children;
}
Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};
Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};
Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};
Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};
Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};
Header.Search = function HeaderSearch({searchTerm, setSearchTerm, children, ...restProps }) {
  const [searchActive, setSearchActive] = useState(false);
  return (<Search {...restProps}>
  <SearchIcon onClick={() => setSearchActive((searchActive) => !searchActive)} data-testid="search-click" >
  <img src="/images/icons/search.png" alt="search"/>
  </SearchIcon>
  <SearchInput value={searchTerm} onChange={({target}) => setSearchTerm(target.value)}
   placeholder="Search films and series" active={searchActive} data-testid="search-input">
   </SearchInput>
  </Search>);
};
Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`}/>;
};
Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};
Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};
Header.Feature = function HeaderFeature({children, ...restProps}){
    return <Feature {...restProps}>{children}</Feature>;
}
Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};
Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};
Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
      <ReactRouterLink to={to}>
      <Logo {...restProps} />
      </ReactRouterLink>

  );
};

