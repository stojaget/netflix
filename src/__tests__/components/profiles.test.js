import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../components';
describe('<Profiles />', () => {
    it('renders <Profiles /> with the populated data', () => {
         const {container, getByText, getByTestId} = render(
              <Profiles>
          <Profiles.Title>Who's watching?</Profiles.Title>
          <Profiles.List>
          <Profiles.User onClick={() => {}}>
          <Profiles.Picture src="/images/users/1.png" data-testid="profile-picture" />
          <Profiles.Name>darac</Profiles.Name>
          </Profiles.User>
          </Profiles.List>
        </Profiles>
         );
         expect(getByText("Who's watching?")).toBeTruthy();
 expect(getByTestId('profile-picture')).toBeTruthy();
          expect(getByText('darac')).toBeTruthy();
             expect(container.firstChild).toMatchSnapshot();
    });
     it('renders <Profiles /> with the populated data but misc profile image', () => {
         const {container, getByText, getByTestId} = render(
              <Profiles>
          <Profiles.Title>Who's watching?</Profiles.Title>
          <Profiles.List>
          <Profiles.User onClick={() => {}}>
          <Profiles.Picture src="/images/users/1.png" data-testid="profile-picture-misc" />
          <Profiles.Name>darac</Profiles.Name>
          </Profiles.User>
          </Profiles.List>
        </Profiles>
         );
         expect(getByText("Who's watching?")).toBeTruthy();
 expect(getByTestId('profile-picture-misc')).toBeTruthy();
          expect(getByText('darac')).toBeTruthy();
             expect(container.firstChild).toMatchSnapshot();
    });
});
