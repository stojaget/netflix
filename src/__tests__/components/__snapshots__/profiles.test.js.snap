// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders <Profiles /> with the populated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 glIXWo"
>
  <h1
    class="profiles__Title-bxtkkv-1 kxHuEI"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 kUSmfb"
  >
    <li
      class="profiles__Item-bxtkkv-5 fFKdrN"
    >
      <img
        class="profiles__Picture-bxtkkv-4 izXQyg"
        data-testid="profile-picture"
        src="/images/users//images/karl.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 ldskSv"
      >
        darac
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders <Profiles /> with the populated data but 1`] = `
<div
  class="profiles__Container-bxtkkv-0 glIXWo"
>
  <h1
    class="profiles__Title-bxtkkv-1 kxHuEI"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 kUSmfb"
  >
    <li
      class="profiles__Item-bxtkkv-5 fFKdrN"
    >
      <img
        class="profiles__Picture-bxtkkv-4 izXQyg"
        data-testid="profile-picture"
        src="/images/users//images/karl.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 ldskSv"
      >
        darac
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders <Profiles /> with the populated data but misc profile image 1`] = `
<div
  class="profiles__Container-bxtkkv-0 glIXWo"
>
  <h1
    class="profiles__Title-bxtkkv-1 kxHuEI"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 kUSmfb"
  >
    <li
      class="profiles__Item-bxtkkv-5 fFKdrN"
    >
      <img
        class="profiles__Picture-bxtkkv-4 izXQyg"
        data-testid="profile-picture-misc"
        src="/images/users//images/karl.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 ldskSv"
      >
        darac
      </p>
    </li>
  </ul>
</div>
`;
